### PR TITLE
Adding ability to specify d3 curve function name as a line series prop

### PR DESCRIPTION
**Pull request by skastel**
In order to make smoothed line series, a user should be able to specify the name of one of the many smoothing functions that d3-shape provides: https://github.com/d3/d3-shape#curves

### CODE REVIEW

**Comment by mcnuttandrew** [comment]
Hey @skastel 
This is super cool! Can you add an example to the examples page of this in use?

**Comment by skastel** [comment]
@mcnuttandrew Done added some smoothed lines in the examples page including on the main complex chart.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-vis",
-  "version": "0.6.5",
+  "version": "0.6.6",
```
Comment (chrisirhc) [inline package.json]: We try to keep version bumps outside of PRs. That way, we can keep commit history readable/clean, and the version bump/publishing process tied closely together.

File: docs/xy-plot.md [hunk 2]
```diff
@@ -363,6 +363,11 @@ Default: `auto`
 The way to align the hint inside the chart. When `auto` is set the hint is trying to stay inside the bounding box of the chart.
 Set the hint to `topleft` if you want to see a "conventional" hint alignment.
 
+#### curve (optional)
+Type: `string`
+Default: `null`__
```
Comment (chrisirhc) [inline docs/xy-plot.md]: trailing `__` ?

File: README.md [hunk 3]
```diff
@@ -83,6 +83,10 @@ npm test
 
 ## Change log
 
+### 0.6.6
```
Comment (chrisirhc) [inline README.md]: Let's add `(unreleased)` here, so we can remove it when we release, and we can append to the change log later.

File: docs/xy-plot.md (hunk 2 above)
Comment (skastel) [inline docs/xy-plot.md]: Oops, fixed.

File: package.json (hunk 1 above)
Comment (skastel) [inline package.json]: Reverted.

File: README.md (hunk 3 above)
Comment (skastel) [inline README.md]: Done.

File: src/lib/plot/series/line-series.js [hunk 4]
```diff
@@ -36,15 +36,27 @@ const STROKE_STYLES = {
...
+  strokeStyle: React.PropTypes.oneOf(Object.keys(STROKE_STYLES)),
+  curve: React.PropTypes.string
 };
 
 class LineSeries extends AbstractSeries {
+
+  _renderLine(data, x, y, curve) {
+    let line = d3Shape.line();
+    if (curve !== null && d3Shape[curve]) {
+      line = line.curve(d3Shape[curve]);
```
Comment (chrisirhc) [inline src/lib/plot/series/line-series.js]: Was looking through the curve documentation. Any thoughts on how to specify `var line = d3.line().curve(d3.curveCatmullRom.alpha(0.5));` with the curve prop?

Comment (skastel) [inline src/lib/plot/series/line-series.js]: How about allowing users to import d3-shape and configure and submit their own curve function:
```javascript
if (curve !== null) {
  if (typeof curve === 'string' && d3Shape[curve]) {
    line = line.curve(d3Shape[curve]);
  } else if (typeof curve === 'function') {
    line = line.curve(curve);
  }
}
```

Comment (chrisirhc) [inline src/lib/plot/series/line-series.js]: Sounds good 👍

Comment (skastel) [inline src/lib/plot/series/line-series.js]: Ok, updated to support functions as well as strings, also updated the docs accordingly.

**Pull request merged**